fe/fe-core/src/main/java/com/starrocks/catalog/ExternalCatalog.java

```diff
@@ -26,15 +26,29 @@ public ExternalCatalog(long id, String name, String comment, Map<String, String>
         Preconditions.checkNotNull(config.get(CATALOG_TYPE));
     }
 
-    // database uuid format: external_catalog_name.db_name
+    // old database uuid format: external_catalog_name.db_name
+    // new database uuid format: db_name
     public static String getDbNameFromUUID(String uuid) {
         // To be in compatible with code before external table privilege is supported
         return uuid.contains(".") ? uuid.split("\\.")[1] : uuid;
     }
 
-    // table uuid format: external_catalog_name.db_name.table_name.creation_time
+    // old table uuid format: external_catalog_name.db_name.table_name.creation_time
+    // new table uuid format: table_name
     public static String getTableNameFromUUID(String uuid) {
         // To be in compatible with code before external table privilege is supported
         return uuid.contains(".") ? uuid.split("\\.")[2] : uuid;
     }
+
+    // old table uuid format: external_catalog_name.db_name.table_name.creation_time
+    // new table uuid format: table_name
+    public static String getCompatibleTableUUID(String uuid) {
```

Comment:
Looks like this will also be called by internal catalog table, and it implys that internal catalog table doesn't contain `dot`. So either move these two methods to a higher level class, like `Catalog` or distinguish whether it's internal or external catalog at the caller and make these methods only called by external table. For the first case, you need to change the comments also, i.e. what's the uuid format of internal table. So i recommend the second fix.